core/model/src/main/java/org/eclipse/rdf4j/model/Literal.java

```diff
@@ -47,6 +48,10 @@
 	 */
 	public IRI getDatatype();
 
+	default public XmlDatatypeEnum getXmlDatatypeEnum() {
```

Comment:
I agree with you that use of Optional is better here (especially if it makes no difference in performance) - it will also make some of your usage code a bit more fluent. Btw the `public` modifier is redundant here (and on everything else in this interface as well I guess).

Reply:
Naming: as with the class name, I dislike having the type in the method name. Can we call the method `getXmlDatatype` instead?

Reply:
I've gone with `getXsdDatatype()` which aligns with having `Datatype` be an enum within the `XSD` class.

Reply:
I've also moved to using `Optional`.